src/fparser/parsefortran.py

```diff
@@ -118,22 +118,27 @@ def put_item(self, item):
         return
 
     def parse(self):
+        '''Parses the program specified in the reader object.'''
         if self.block is not None:
             return
         try:
-            block = self.block = BeginSource(self)
+            self.block = BeginSource(self)
+        except AnalyzeError:
+            raise
         except KeyboardInterrupt:
             raise
         except:
             reader = self.reader
             while reader is not None:
-                message = reader.format_message('FATAL ERROR',
-                                                'while processing line',
-                                                reader.linecount, reader.linecount)
+                message = \
+                    reader.format_message('FATAL ERROR',
+                                          'while processing line',
```

Comment:
This code is not covered but as it has only been tidied for pep8 I won't request it.